config/config-server/src/main/resources/cruise-config.xsd

```diff
@@ -848,7 +848,7 @@
   <xsd:simpleType name="labelType">
     <xsd:restriction base="xsd:string">
       <xsd:pattern
-          value="(([a-zA-Z0-9_\-.!~*'()#:])*[$#]\{[a-zA-Z0-9_\-.!~*'()#:]+(\[:(\d+)\])?\}([a-zA-Z0-9_\-.!~*'()#:])*)+"/>
+          value="(([a-zA-Z0-9_\-.!~*'()#:])*[$#]\{[$]*[a-zA-Z0-9_\-.!~*'()#:]+(\[:(\d+)\])?\}([a-zA-Z0-9_\-.!~*'()#:])*)+"/>
```

Comment:
This is a scary regex. Do we really need it here, if it's going to be checked later in the PipelineConfig? Might be a concern outside of this PR, though.

Reply:
It is scary regex. I think the possibility of removing it from the schema is maybe a little outside the scope here.